TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/workflow/services/script_runner.py

```diff
@@ -0,0 +1,391 @@
+        index_js_path = self.script_dir / "index.js"
+
+        if index_sh_path.exists():
+            cmd = [
+                str(index_sh_path),
+                "--user-config", "-",
+                "--silent"
+            ]
+        elif index_js_path.exists():
+            cmd = [
+                "node",
+                str(index_js_path),
+                "--user-config", "-",
+                "--silent"
+            ]
+        else:
+            raise FileNotFoundError(f"Config processor not found: {index_js_path} or {index_sh_path}")
+
+        # Add prefix if provided
+        if prefix:
+            cmd.extend(["--prefix", prefix])
+
+        logger.debug(f"Running config processor for initialization: {' '.join(cmd)}")
+        logger.debug(f"Input config length: {len(config_data)} bytes")
+
+        try:
+            result = subprocess.run(
+                cmd,
+                input=config_data,
+                capture_output=True,
+                text=True,
+                check=True,
+                cwd=str(self.script_dir)
+            )
+
+            # The output should be the prefix (either provided or auto-generated)
+            output_prefix = result.stdout.strip()
+            logger.info(f"Workflow initialized with prefix: {output_prefix}")
+            return output_prefix
+
+        except subprocess.CalledProcessError as e:
+            logger.error(f"Workflow initialization failed with exit code {e.returncode}")
+            logger.error(f"stderr: {e.stderr}")
+            raise
+
+    def submit_workflow(
+        self,
+        config_data: str,
+        prefix: str,
+        namespace: str = "ma"
+    ) -> Dict[str, Any]:
+        """Submit workflow using config processor submission script.
+
+        Calls: {script_dir}/createMigrationWorkflowFromUserConfiguration.sh <temp_file>
+
+        The script creates the workflow in Kubernetes and returns workflow information.
+
+        Args:
+            config_data: User configuration YAML as string
+            prefix: Workflow prefix (from init_workflow)
+            namespace: Kubernetes namespace (default: "ma")
+
+        Returns:
+            Dict with workflow_name, workflow_uid, and namespace
+
+        Raises:
+            FileNotFoundError: If submission script doesn't exist
+            subprocess.CalledProcessError: If script fails
+            ValueError: If script output cannot be parsed
+        """
+        logger.info(f"Submitting workflow with prefix: {prefix}, namespace: {namespace}")
+
+        script_path = self.script_dir / "createMigrationWorkflowFromUserConfiguration.sh"
+
+        if not script_path.exists():
+            raise FileNotFoundError(f"Workflow submission script not found: {script_path}")
+
+        # Create temporary file with config data
+        with tempfile.NamedTemporaryFile(mode='w', suffix='.yaml', delete=False) as temp_file:
+            temp_file.write(config_data)
```

Comment:
FWIW, both the initializer node script & the shell script can handle stdin via '-', IIRC